runtime/transformations/components/filters/correct_time_intervals_with_fixed_values.py

```diff
@@ -0,0 +1,237 @@
+
+## Outputs
+- **corrected** (Pandas Series):
+    The Series with corrected values.
+
+## Details
+- If a data point lies within multiple time intervals, it will be changed to the `correction_value`
+of the last time interval in `time_interval_dict`.
+- Raises `ComponentInputValidationException`:
+    - If the index of `series` is not a DateTimeIndex.
+    - If `list_of_time_intervals` contains any invalid entries.
+
+## Examples
+
+Example json input for a call of this component is:
+```
+{   "series": {
+        "2020-01-01T01:15:27.000Z": 42.2,
+        "2020-01-03T08:20:03.000Z": 18.7,
+        "2020-01-03T08:20:04.000Z": 25.9
+    },
+    "list_of_time_intervals": [
+        {
+            "start": "2020-01-01T01:15:27.000Z",
+            "end": "2020-01-01T01:15:27.000Z",
+            "start_inclusive": true,
+            "end_inclusive": true,
+            "correction_value": 37
+        },
+        {
+            "start": "2020-01-03T08:20:02.000Z",
+            "end": "2020-01-03T08:20:04.000Z",
+            "start_inclusive": false,
+            "end_inclusive": false,
+            "correction_value": 42.0
+        }
+    ]
+}
+```
+Output of the above call is:
+```
+{
+    "corrected":{
+        "__hd_wrapped_data_object__":"SERIES",
+        "__metadata__":{},
+        "__data__":{
+            "2020-01-01T01:15:27.000Z":37.0,
+            "2020-01-03T08:20:03.000Z":42.0,
+            "2020-01-03T08:20:04.000Z":25.9
+        }
+    }
+}
+```
+"""
+
+import pandas as pd
+from pydantic import BaseModel, ValidationError, root_validator
+
+from hetdesrun.runtime.exceptions import ComponentInputValidationException
+
+
+class TimeInterval(BaseModel):
+    start: str
+    end: str
+    correction_value: float
+    start_inclusive: bool = True
+    end_inclusive: bool = True
+
+    @root_validator()
+    def verify_value_ranges(cls, values: dict) -> dict:
+        start = values["start"]
+        start_inclusive = values["start_inclusive"]
+        end = values["end"]
+        end_inclusive = values["end_inclusive"]
+        if end < start:
+            raise ValueError(
+                "To be valid, a time interval must be non-empty, i.e. the start timestamp "
+                "may not be bigger than the end timestamp."
+            )
+        if (start_inclusive is False or end_inclusive is False) and not (start < end):
+            raise ValueError(
+                "To be valid, a time interval must be non-empty, i.e the start timestamp must be "
+                "smaller than the end timestamp."
+            )
+        return values
+
+
+# ***** DO NOT EDIT LINES BELOW *****
+# These lines may be overwritten if component details or inputs/outputs change.
+COMPONENT_INFO = {
+    "inputs": {
+        "series": {"data_type": "SERIES"},
+        "list_of_time_intervals": {"data_type": "ANY"},
+    },
+    "outputs": {
+        "corrected": {"data_type": "SERIES"},
+    },
+    "name": "Correct time intervals with fixed values",
```

Comment:
All words longer than 3 letters should start with a capital letter in the title.